File: src/spirite/base/pen/behaviors/MovingNodeBehavior.java
package spirite.base.pen.behaviors;

import spirite.base.image_data.GroupTree.Node;
import spirite.base.pen.Penner;

public class MovingNodeBehavior extends StateBehavior {
	final Node node;
	
	public MovingNodeBehavior( Penner penner, Node node) {
		super(penner);
		this.node = node;
	}
	@Override public void start() {}
	@Override public void onTock() {}
	@Override
	public void onMove() {
		if( node != null && (this.penner.oldX != this.penner.x || this.penner.oldY != this.penner.y))
			node.setOffset( node.getOffsetX() + (this.penner.x - this.penner.oldX), 
							 node.getOffsetY() + (this.penner.y - this.penner.oldY));
	}
}

File: src/main/kotlin/spirite/base/pen/behaviors/MovingNodeBehavior.kt
package spirite.base.pen.behaviors

import spirite.base.imageData.groupTree.Node
import spirite.base.pen.Penner


class MovingNodeBehavior(
        penner: Penner,
        val node: Node
)
    : PennerBehavior(penner)
{
    override fun onStart() {}
    override fun onTock() {}
    override fun onMove() {
        if( penner.oldX != penner.x || penner.oldY != penner.y) {
            node.x += penner.x - penner.oldX
            node.y += penner.y - penner.oldY
        }
    }
}